src/plugins/unified_doc_viewer/public/components/doc_viewer_logs_overview/logs_overview.tsx

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import { DocViewRenderProps } from '@kbn/unified-doc-viewer/types';
+import { getDocumentOverview } from '@kbn/discover-utils';
+import { EuiSpacer } from '@elastic/eui';
+import { LogsOverviewHeader } from './logs_overview_header';
+import { LogsOverviewHighlights } from './logs_overview_highlights';
+import { FieldActionsProvider } from '../../hooks/use_field_actions';
+import { getUnifiedDocViewerServices } from '../../plugin';
+
+export function LogsOverview({
+  columns,
+  dataView,
+  hit,
+  filter,
+  onAddColumn,
+  onRemoveColumn,
+}: DocViewRenderProps) {
+  const { fieldFormats } = getUnifiedDocViewerServices();
+  const parsedDoc = getDocumentOverview(hit, { dataView, fieldFormats });
```

Comment:
I didn't go for memoization here since for most of the interactions that could cause a rerender such as filtering for a field, the `hit` property would change (contains `highlights`) and consequently nullify the memoization anyway, which seemed a de-optimization.

As a side note, I triggered multiple rerenders with 6x CPU throttling, and the function execution is always in a 0.1-0.3ms range.